### PR TITLE
Registry docs

### PR DESCRIPTION
**Pull request by lynxplay**

### CODE REVIEW

File: docs/paper/dev/api/registries.mdx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+Mutating registries is done via the
+<Javadoc
+  name={"io.papermc.paper.plugin.lifecycle.event.LifecycleEventManager"}>LifecycleEventManager</Javadoc>.
+See the [Lifecycle Events](./lifecycle.mdx) page for more information.
+:::
+
+The general entrypoint for mutating registries is
+the <Javadoc name={"io.papermc.paper.registry.event.RegistryEvents"}>RegistryEvents</Javadoc> type,
+which provides an entry point for each registry that can be modified.
+Modification of a registries can take two different forms.
```
Comment (kennytv) [inline docs/paper/dev/api/registries.mdx:110]:
```suggestion
Modification of a registry can take two different forms.
```

File: docs/paper/dev/api/registries.mdx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+To retrieve elements from a registry, their respective keys can be used.
+The API defines two types of keys.
+<ol>
+  <li>
+    `net.kyori.adventure.key.Key` represent a namespace and a key.
+  </li>
+  <li>
+    <Javadoc name={"io.papermc.paper.registry.TypedKey"}>TypedKey</Javadoc> wrap an adventure key
+    but also including the <Javadoc name={"io.papermc.paper.registry.TypedKey#registryKey()"}>key of
+    the registry</Javadoc> the
```
Comment (kennytv) [inline docs/paper/dev/api/registries.mdx:35]:
```suggestion
    `net.kyori.adventure.key.Key` represents a namespace and a key.
  </li>
  <li>
    <Javadoc name={"io.papermc.paper.registry.TypedKey"}>TypedKey</Javadoc> wraps an adventure key,
    but also includes the <Javadoc name={"io.papermc.paper.registry.TypedKey#registryKey()"}>key of
    the registry</Javadoc> the
```
singular sounds better here

File: docs/paper/dev/api/registries.mdx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    // Arbitrary keys of enchantments to store in the key set.
+    EnchantmentKeys.CHANNELING,
+    TypedKey.create(RegistryKey.ENCHANTMENT, Key.key("papermc:softspoon"))
+);
+```
+
+A <Javadoc name={"io. papermc. paper. registry. tag.Tag"}>`Tag`</Javadoc> follows up the concept
+of a <Javadoc name={"io.papermc.paper.registry.set.RegistryKeySet"}>`RegistryKeySet`</Javadoc>
+but is itself named and can hence be referenced.
+A list of vanilla tags can be found <a href="https://minecraft.wiki/w/Tag#Java_Edition_2">on the minecraft wiki</a>.
```
Comment (powercasgamer) [inline docs/paper/dev/api/registries.mdx:89]:
```suggestion
A list of Vanilla tags can be found <a href="https://minecraft.wiki/w/Tag#Java_Edition_2">on the minecraft wiki</a>.
```
```suggestion
A list of vanilla tags can be found <a href="https://minecraft.wiki/w/Tag#Java_Edition_2">on the minecraft wiki</a>.
```

File: docs/paper/dev/api/registries.mdx [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+The general entrypoint for mutating registries is
+the <Javadoc name={"io.papermc.paper.registry.event.RegistryEvents"}>RegistryEvents</Javadoc> type,
+which provides an entry point for each registry that can be modified.
+Modification of a registry can take two different forms.
+
+### Create new entries
+
+Creating new entries is done via the <Javadoc name={"io.papermc.paper.registry.event.RegistryEventProvider#freeze"}>`freeze` lifecycle event</Javadoc>
+on the respective registries.
+The freeze event is called right before a registries content is frozen in-place, meaning all vanilla entries are registered.
```
Comment (powercasgamer) [inline docs/paper/dev/api/registries.mdx:119]:
```suggestion
The freeze event is called right before a registries content is frozen in-place, meaning all Vanilla entries are registered.
```
```suggestion
The freeze event is called right before a registries content is frozen in-place, meaning all vanilla entries are registered.
```

File: docs/paper/dev/api/registries.mdx [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+                    .activeSlots(EquipmentSlotGroup.ANY)
+            );
+        }));
+    }
+}
+```
+
+### Modifying existing entries
+
+Modification of existing entries is useful for plugins that aim to change the way vanilla entries
```
Comment (powercasgamer) [inline docs/paper/dev/api/registries.mdx:150]:
```suggestion
Modification of existing entries is useful for plugins that aim to change the way Vanilla entries
```
```suggestion
Modification of existing entries is useful for plugins that aim to change the way vanilla entries
```

**Review by powercasgamer — CHANGES_REQUESTED** [review]

File: docs/paper/dev/api/registries.mdx [hunk 6]
```diff
@@ -0,0 +1,165 @@
+---
+slug: /dev/registries
+description: A guide to registries and their modification on paper
```
Comment (zlataovce) [inline docs/paper/dev/api/registries.mdx:3]:
```suggestion
description: A guide to registries and their modification on Paper.
```

File: docs/paper/dev/api/registries.mdx [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+---
+
+# Registries
+
+:::danger[Experimental]
+The Registry API and anything that uses it is currently experimental and may change in the future.
+:::
+
+## What is a registry ?
```
Comment (zlataovce) [inline docs/paper/dev/api/registries.mdx:12]:
```suggestion
## What is a registry?
```

File: docs/paper/dev/api/registries.mdx [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+
+In the context of Minecraft, a registry holds onto a set of values of the same type, identifying
+each by a key. An example of such a registry would be the <Javadoc name={"org.bukkit.Registry#ITEM"}>ItemType registry</Javadoc> which holds all known item types.
+Registries are available via the <Javadoc name={"io.papermc.paper.registry.RegistryAccess"}>RegistryAccess</Javadoc>.
```
Comment (zlataovce) [inline docs/paper/dev/api/registries.mdx:16]:
```suggestion
Registries are available via the <Javadoc name={"io.papermc.paper.registry.RegistryAccess"}>RegistryAccess</Javadoc> class.
```
or
```suggestion
Registries are available via <Javadoc name={"io.papermc.paper.registry.RegistryAccess"}>RegistryAccess</Javadoc>.
```

File: docs/paper/dev/api/registries.mdx [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+
+While a large portion of registries are defined by the server and client independently, more and
+more are defined by the server and sent to the client while joining the server.
+This enables the server, and to that extend plugins, to define custom content for both itself and
```
Comment (zlataovce) [inline docs/paper/dev/api/registries.mdx:20]:
```suggestion
This enables the server, and to that extent plugins, to define custom content for both itself and
```

File: docs/paper/dev/api/registries.mdx [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+### Retrieving values from a registry
+
+To retrieve elements from a registry, their respective keys can be used.
+The API defines two types of keys.
+<ol>
+  <li>
+    `net.kyori.adventure.key.Key` represents a namespace and a key.
+  </li>
+  <li>
+    <Javadoc name={"io.papermc.paper.registry.TypedKey"}>TypedKey</Javadoc> wraps an adventure key,
```
Comment (zlataovce) [inline docs/paper/dev/api/registries.mdx:33]:
```suggestion
    <Javadoc name={"io.papermc.paper.registry.TypedKey"}>TypedKey</Javadoc> wraps an Adventure key,
```

File: docs/paper/dev/api/registries.mdx [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+    but also includes the <Javadoc name={"io.papermc.paper.registry.TypedKey#registryKey()"}>key of
+    the registry</Javadoc> the
+    <Javadoc name={"io.papermc.paper.registry.TypedKey"}>TypedKey</Javadoc> belongs to.
+  </li>
+</ol>
+An example of retrieving the sharpness enchantment using
```
Comment (zlataovce) [inline docs/paper/dev/api/registries.mdx:39]:
```suggestion
An example of retrieving the `Sharpness` enchantment using
```

File: docs/paper/dev/api/registries.mdx [hunk 12]
```diff
@@ -0,0 +1,165 @@
...
+
+// Same as above, but using generated typed keys.
+// Only Vanilla entries have generated keys, for custom entries, the above method must be used.
+final Enchantment enchantment = enchantmentRegistry.getOrThrow(EnchantmentKeys.SHARPNESS);
+```
+
+### Referencing registry values
+
+Referencing entries in a registry is easier said then done.
+While for most cases a plain Collection of the values might suffice, alternative approaches are
```
Comment (zlataovce) [inline docs/paper/dev/api/registries.mdx:62]: presuming it's a reference to the Java stdlib class, `Collection` could be encased in backticks and/or linked to the corresponding Javadoc

File: docs/paper/dev/api/registries.mdx [hunk 13]
```diff
@@ -0,0 +1,165 @@
...
+// Create a new registry key set that holds a collection enchantments
+final RegistryKeySet<@NotNull Enchantment> bestEnchantments = RegistrySet.keySet(
+    RegistryKey.ENCHANTMENT,
+    // Arbitrary keys of enchantments to store in the key set.
+    EnchantmentKeys.CHANNELING,
+    TypedKey.create(RegistryKey.ENCHANTMENT, Key.key("papermc:softspoon"))
+);
+```
+
+A <Javadoc name={"io. papermc. paper. registry. tag.Tag"}>`Tag`</Javadoc> follows up the concept
```
Comment (zlataovce) [inline docs/paper/dev/api/registries.mdx:86]:
```suggestion
A <Javadoc name={"io.papermc.paper.registry.tag.Tag"}>`Tag`</Javadoc> follows up the concept
```

File: docs/paper/dev/api/registries.mdx [hunk 14]
```diff
@@ -0,0 +1,165 @@
...
+of a <Javadoc name={"io.papermc.paper.registry.set.RegistryKeySet"}>`RegistryKeySet`</Javadoc>
+but is itself named and can hence be referenced.
+A list of Vanilla tags can be found <a href="https://minecraft.wiki/w/Tag#Java_Edition_2">on the minecraft wiki</a>.
```
Comment (zlataovce) [inline docs/paper/dev/api/registries.mdx:89]:
```suggestion
A list of Vanilla tags can be found [on the Minecraft wiki](https://minecraft.wiki/w/Tag#Java_Edition_2).
```

File: docs/paper/dev/api/registries.mdx [hunk 15]
```diff
@@ -0,0 +1,165 @@
...
+
+## Mutating registries
+
+Beyond plain reading access to registries, paper also offers a way for plugins to modify registries.
+
+:::warning
+Mutating registries needs to be done during the servers bootstrap phase.
```
Comment (zlataovce) [inline docs/paper/dev/api/registries.mdx:96]:
```suggestion
Mutating registries needs to be done during the server's bootstrap phase.
```

File: docs/paper/dev/api/registries.mdx [hunk 16]
```diff
@@ -0,0 +1,165 @@
...
+```
+
+A <Javadoc name={"io. papermc. paper. registry. tag.Tag"}>`Tag`</Javadoc> follows up the concept
+of a <Javadoc name={"io.papermc.paper.registry.set.RegistryKeySet"}>`RegistryKeySet`</Javadoc>
+but is itself named and can hence be referenced.
+A list of Vanilla tags can be found <a href="https://minecraft.wiki/w/Tag#Java_Edition_2">on the minecraft wiki</a>.
+
+## Mutating registries
+
+Beyond plain reading access to registries, paper also offers a way for plugins to modify registries.
```
Comment (zlataovce) [inline docs/paper/dev/api/registries.mdx:93]:
```suggestion
Beyond plain reading access to registries, Paper also offers a way for plugins to modify registries.
```

File: docs/paper/dev/api/registries.mdx [hunk 17]
```diff
@@ -0,0 +1,165 @@
...
+
+:::warning
+Mutating registries needs to be done during the servers bootstrap phase.
+As such, this section is only applicable to [Paper plugins](../getting-started/paper-plugins.mdx).
+
+**Exceptions** thrown by plugins during this phase will cause the server to shutdown before loading
```
Comment (zlataovce) [inline docs/paper/dev/api/registries.mdx:99]:
```suggestion
**Exceptions** thrown by plugins during this phase will cause the server to shutdown before loading,
```

File: docs/paper/dev/api/registries.mdx [hunk 18]
```diff
@@ -0,0 +1,165 @@
...
+as missing values or modifications to the registries would otherwise cause data loss.
+:::
+
+:::note
+Mutating registries is done via the
```
Comment (zlataovce) [inline docs/paper/dev/api/registries.mdx:104]:
```suggestion
Mutating registries is done via
```

File: docs/paper/dev/api/registries.mdx [hunk 19]
```diff
@@ -0,0 +1,165 @@
...
+:::
+
+The general entrypoint for mutating registries is
+the <Javadoc name={"io.papermc.paper.registry.event.RegistryEvents"}>RegistryEvents</Javadoc> type,
+which provides an entry point for each registry that can be modified.
+Modification of a registry can take two different forms.
+
+### Create new entries
+
+Creating new entries is done via the <Javadoc name={"io.papermc.paper.registry.event.RegistryEventProvider#freeze"}>`freeze` lifecycle event</Javadoc>
```
Comment (zlataovce) [inline docs/paper/dev/api/registries.mdx:117]:
```suggestion
Creating new entries is done via the <Javadoc name={"io.papermc.paper.registry.event.RegistryEventProvider#freeze()"}>`freeze` lifecycle event</Javadoc>
```

File: docs/paper/dev/api/registries.mdx [hunk 20]
```diff
@@ -0,0 +1,165 @@
...
+on the respective registries.
+The freeze event is called right before a registries content is frozen in-place, meaning all Vanilla entries are registered.
```
Comment (zlataovce) [inline docs/paper/dev/api/registries.mdx:119]:
```suggestion
The freeze event is called right before a registry's content is frozen in-place, meaning all Vanilla entries are registered.
```

File: docs/paper/dev/api/registries.mdx [hunk 21]
```diff
@@ -0,0 +1,165 @@
...
+            );
+        }));
+    }
+}
+```
+
+### Modifying existing entries
+
+Modification of existing entries is useful for plugins that aim to change the way Vanilla entries
+behave. For this the <Javadoc name={"io.papermc.paper.registry.event.RegistryEventProvider#entryAdd"}>`entryAdd` lifecycle event</Javadoc>.
```
Comment (zlataovce) [inline docs/paper/dev/api/registries.mdx:151]:
```suggestion
behave. For this, use the <Javadoc name={"io.papermc.paper.registry.event.RegistryEventProvider#entryAdd()"}>`entryAdd` lifecycle event</Javadoc>.
```

File: docs/paper/dev/api/registries.mdx [hunk 22]
```diff
@@ -0,0 +1,165 @@
...
+The event is called for <u>*any*</u> entry added to a registry, however the API provides an easy way to target a specific entry for modification.
```
Comment (zlataovce) [inline docs/paper/dev/api/registries.mdx:152]:
```suggestion
The event is called for _\*any\*_ entry added to a registry, however the API provides an easy way to target a specific entry for modification.
```

File: docs/paper/dev/api/registries.mdx [hunk 23]
```diff
@@ -0,0 +1,165 @@
...
+The following example shows how to increase the maximum level of the sharpness enchantment.
```
Comment (zlataovce) [inline docs/paper/dev/api/registries.mdx:153]:
```suggestion
The following example shows how to increase the maximum level of the `Sharpness` enchantment.
```

**Review by kennytv — APPROVED** [review]

File: docs/paper/dev/api/registries.mdx (hunk 18 above)
Comment (lynxplay) [reply docs/paper/dev/api/registries.mdx:104]: No

**Review by kashike — APPROVED** [review]

**Pull request merged**